ui-test/src/UiTestUtilities.ts

```diff
@@ -0,0 +1,298 @@
+import Configuration from './Configuration';
```

Comment:
I'm a big fan of page object pattern. The idea is to group page-specific methods into a class that represents a page and class instance can be created by navigating to the page so that test look as following:

```javascript

// initializes web driver and returns instance of `Navigation` class that holds`WebDriver` instance
const navigation = utils.init();
// navigates to `/applications` page, waits for navigation to complete and returns `ApplicationsList` class
const appListPage = navigation.goToAppList();
// clicks new app buttons and returns `NewApplicationDialog` that holds link to sliding panel element
const newAppDialog = await appListPage.newAppButtonClick();
// locates app name element and sends 'myApp' keys
await newAppDialog.setAppName('myApp');
...

```

`ApplicationsList:`

```typescript
export class ApplicationsList {
  panel: WebElement;
  public async setAppName(val: string) {
    findUiElement(Const.CREATE_APPLICATION_FIELD_APP_NAME);
    await appNameField.sendKeys(appName);
  }
}
```

This approach was very helpful in one of my previous projects and helped us to effectively structure test utility functions.

Reply:
Hi @alexmt, what's the naming convention for file names in ArgoCD?  For example, https://github.com/argoproj/argo-cd/blob/d2295d45d0f42b5c8f8e77715364fda13b10acd4/ui/src/app/shared/components/version-info/version-info-panel.tsx#L13 , the file name is `version-info-panel.tsx` and the class name defined is `VersionPanel`.  Although, not all files contain classes, is probably one reason why VersionPanel.tsx is not used (I'm thinking Java, of course).